packages/jupyterlab-manager/src/manager.ts

```diff
@@ -156,7 +160,11 @@ class WidgetManager extends ManagerBase<Widget> implements IDisposable {
   _handleKernelStatusChange(args: Kernel.Status) {
     switch (args) {
     case 'connected':
-      this.restoreWidgets(this._context.model);
+      // Only restore if our initial restore at construction is finished
```

Comment:
I think this logic is faulty:

`_restoredStatus` is also set to `false` during a disconnect/reconnect cycle, and we only want this prevention during the initial construction (as this comment says). I think the right was to do this is to not connect directly in the ctor, but rather only connect after the inital restored signal is sent. Actually, ideally the initial restore would have been a promise so that we also connect if the initial restore throws an exception.

Reply:
We need to connect right away, since there are other things happening such as calling the parent disconnect method. I've added a new variable to keep track of the initial restore, though.